enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +35,294 @@ have the ability to apply user defined tags to many resources created by OpenShi
+
+### Limitations
+
+- User-defined tags for PV volumes cannot be updated but only set during creation  of volume using `--extra-tags`.Present CSI spec does not support `UpdateVolume` to update volume metadata.
+   Restarting csi driver to initiate multiple `CreateVolume` will depend on underlying storage layer idempotency implementation for the operation. It is recommended to use a interface from CSI spec which enforces idempotency guarantees for volume metadata updates.
+   In case of AWS EC2  `CreateVolume`, returns `IdempotentParameterMismatch` error when identical request is not found for the same client token.
+   Also, the CSI spec `https://github.com/container-storage-interface/spec/blob/master/spec.md` mentions grpc error code
+   `6 ALREADY_EXISTS` for `a volume corresponding to the specified volume name already exists but is incompatible with the specified capacity_range, volume_capabilities, parameters, accessibility_requirements or volume_content_source`.
+- User-defined tags cannot be updated for vpc, security groups, elb, route53, subnet resources as there is no operator managing the resources post installation.
+   User-defined tags cannot be updated to an AWS resource which is not managed by an operator in openshift-* namespace.
 
 ## Proposal
 
-New `experimentalPropagateUserTags` field added to `.platform.aws` of install config to indicate that the user tags should be applied to AWS
+- Existing `experimentalPropagateUserTags` will be renamed to `propagateUserTags` in `.platform.aws` of install config to indicate that the user tags should be applied to AWS
 resources created by in-cluster operators.
+  `propagateUserTags` is mandatory field.
+  `experimentalPropagateUserTags` field will be set for deprecation.
+
+  If `propagateUserTags` is set to true, install validation will fail if there is any tag that starts with `kubernetes.io` or `openshift.io`.
+
+- Add a new field `resourceTags` to `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io` type. Tags included in the
+  `resourceTags` field will be applied to new resources created for the cluster. The `resourceTags` field will be populated by the installer only if the `propagateUserTags` field is true.
+
+  `.spec.platformSpec.aws` is a mutable field and `.status.platformStatus.aws` is immutable.
+  `.status.platformStatus.aws` will have older version tags defined and is required for upgrade case.
+
+- All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api) will apply these tags to all AWS resources they create.
+  Operator must update the AWS resource within (5 minutes + cloud provide API response time).
+  Operator must update only when there is a change in `.spec.platformSpec.aws.resourceTags`.
+  Operator must consider `.status.platformStatus.aws` when AWS resource is created. `.status.platformStatus.aws` can be ignored for user-defined tag update/delete.
```

Comment:
There is no need to sync to status. As spec has the actual values. Sync to status was required in earlier design to identify if the tag was created using Infrastructure CRD or external tool. Identifying the creator tool was done to restrict user from editing the same tag user kv pair using multiple tools. This inherently poses many scenarios of conflicts which will result in user tag kv pair being inconsistent across cluster when applied using Infrastructure CRD. Hence, user will be confused which tool to be used to update tag.

Reply:
I'd counter that with the argument that because we already have some values in the status, it's now confusing to a user because half the values come from spec, half come from status, or have I misunderstood?

It was my understanding that we synced to status so that there's a single source of truth for the rest of the cluster components to read from, I don't remember the status vs spec being used to distinguish between an external tool and the infra CRD?

Are we planning to completely deprecate and remove the status field? Does the status field still serve a purpose?

Reply:
The plan is to completely deprecate and remove the status field.